snuba/query/mark_discover_query.py

```diff
@@ -0,0 +1,12 @@
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class MarkDiscoverQuery(QueryProcessor):
+    """
+    Temporary query processor to support accessing transactions through the events storage.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        query.set_hint("discover")
```

Comment:
I don't think this is a great usage of the idea of query hints. 
If/when we will support them, they would be mostly meant for running tests (like controlling the behavior of the query pipeline within snuplicator, when comparing two solutions) or for guiding some performance optimizations.
In the first case, we would be talking about a temporary test. In those case collecting evidence  should not require a redesign of the pipeline.
In the second case the correctness of the query would not be changed, which means that we could safely ignore the hints and produce the same result.

In this case we would make the correctness of the pipeline depend on the hint to be processed properly. And this hints is used to create a backward dependency from the physical layer to the logical layer, thus breaking the principle that each step of the query processing relies on the query AST and work in isolation.

In order to preserve the architecture of the query pipeline, I would add the condition in the entity query processing for the `events` entity. Now the query is correct in any case. Then we can optimize it in the storage query processing for errors by taking away if possible. The two steps would not depend on each other to produce a correct query. 
I am not concerned about the cpu impact of adding a condition to remove it as we are going to remove this as soon as the error table will be fully rolled out.

I understand that using a higher level hint may look simpler, but I believe that is only because it is the only case. If we started adding dependencies back and forth more frequently, soon understanding the pipeline would become a lot harder. 

I think also this kind of problems will be much smaller once the sdk is used and logical layer is moved to sentry.